matter-labs/era-test-node
# feat: adds hardhat_getAutomine for evm emulator and hardhat ignition usage

CC: @0xVolosnikov 

# What :computer: 
* Hardhat ignition calls `hardhat_getAutomine` for development networks 

# Why :hand:
* This adds support for the hardhat method for testing evm emulator with ignition 

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

To test:

- Checkout evm-emulator contracts commit (e.g. [evm-emulator](https://github.com/matter-labs/era-contracts/tree/integration-stable/evm-emulator))
- Run: `./target/release/era_test_node --dev-system-contracts=local --emulate-evm -d fork <RPC-URL>`

```
npx hardhat ignition deploy ignition/modules/Token.js --network local
✔ Confirm deploy to network local (260)? … yes
Hardhat Ignition 🚀

Deploying [ TokenModule ]

Batch #1
  Executed TokenModule#Token

[ TokenModule ] successfully deployed 🚀

Deployed Addresses

TokenModule#Token - 0x1E5cd1E78CB47fC26D1Cbb3137A88fb30CbEF1Be
```
